tools/docker-network/tests/coreapi_test.go

```diff
@@ -0,0 +1,542 @@
+			testFunc: func(t *testing.T, nodeAlias string) {
+				assetsPerSlot.forEachTransaction(t, func(t *testing.T, transaction *iotago.SignedTransaction, firstAttachmentID iotago.BlockID) {
+					resp, err := d.wallet.Clients[nodeAlias].TransactionIncludedBlock(context.Background(), lo.PanicOnErr(transaction.Transaction.ID()))
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+					require.EqualValues(t, firstAttachmentID, resp.MustID())
+				})
+			},
+		},
+		{
+			name: "Test_TransactionsIncludedBlockMetadata",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				assetsPerSlot.forEachTransaction(t, func(t *testing.T, transaction *iotago.SignedTransaction, firstAttachmentID iotago.BlockID) {
+					resp, err := d.wallet.Clients[nodeAlias].TransactionIncludedBlockMetadata(context.Background(), lo.PanicOnErr(transaction.Transaction.ID()))
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+					require.EqualValues(t, api.BlockStateFinalized, resp.BlockState)
+					require.EqualValues(t, firstAttachmentID, resp.BlockID, "Inclusion BlockID of retrieved transaction does not match: %s != %s", firstAttachmentID, resp.BlockID)
+				})
+			},
+		},
+		{
+			name: "Test_TransactionsMetadata",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				assetsPerSlot.forEachTransaction(t, func(t *testing.T, transaction *iotago.SignedTransaction, firstAttachmentID iotago.BlockID) {
+					resp, err := d.wallet.Clients[nodeAlias].TransactionMetadata(context.Background(), lo.PanicOnErr(transaction.Transaction.ID()))
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+					require.Equal(t, api.TransactionStateFinalized, resp.TransactionState)
+					require.EqualValues(t, resp.EarliestAttachmentSlot, firstAttachmentID.Slot())
+				})
+			},
+		},
+		{
+			name: "Test_Congestion",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				assetsPerSlot.forEachAccountAddress(t, func(
+					t *testing.T,
+					accountAddress *iotago.AccountAddress,
+					commitmentPerNode map[string]iotago.CommitmentID,
+					bicPerNoode map[string]iotago.BlockIssuanceCredits,
+				) {
+					resp, err := d.wallet.Clients[nodeAlias].Congestion(context.Background(), accountAddress, 0)
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+
+					// node allows to get account only for the slot newer than lastCommittedSlot - MCA, we need fresh commitment
+					infoRes, err := d.wallet.Clients[nodeAlias].Info(context.Background())
+					require.NoError(t, err)
+					commitment, err := d.wallet.Clients[nodeAlias].CommitmentBySlot(context.Background(), infoRes.Status.LatestCommitmentID.Slot())
+					require.NoError(t, err)
+
+					resp, err = d.wallet.Clients[nodeAlias].Congestion(context.Background(), accountAddress, 0, commitment.MustID())
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+					// later we check if all nodes have returned the same BIC value for this account
+					bicPerNoode[nodeAlias] = resp.BlockIssuanceCredits
+				})
```

Comment:
We can call assertBICs here, no?

Reply:
This check if all nodes returned the same and iterates over `bicPerNoode map`, so it should be called afterwards